Move user to /plans page when exiting and clearing cart from checkout

paulopmt1 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/99151

Big Sky is a paid feature, and sometimes users will give up at the last minute during the checkout. We need to ensure they still have other plans options when doing that.

## Proposed Changes

* We're moving the user to the /plans page when they select Big Sky and bail from checkout


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* From the [/setup/onboarding](http://calypso.localhost:3000/setup/onboarding), select Big Sky
* Select a paid plan
* In the checkout, bail with any option you want
* Make sure you won't have big sky and yet, have an option of selecting another plan

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

vykes-mac commented:
https://github.com/user-attachments/assets/9727ea19-373e-4972-b11c-92301448b142


It works but the navigation back to plans page is quite jarring at times, don't think it have anything to do with this PR though

vykes-mac commented:
I think we could also do 

```js
if (createWithBigSky && 
config.isEnabled( 'onboarding/big-sky-before-plans' ) && 
isGoalsAtFrontExperiment) {
  destination = '/setup/onboarding/plans';
}
```

inside of the `onboarding.ts` file when going to the checkout, that way we do not redirect to the `site-setup-flow` then back to plans.

paulopmt1 committed:
Fix the exit checkout flow by proper redirecting back to /plans page

paulopmt1 commented:
> What if we don't continue to site-setup but set the checkoutBackUrl to return to /plans when it's big sky related.

Waaay simpler @vykes-mac. I am not sure why I complicated things before, maybe the late Friday event 😆. Thanks, updated!

p-jackson committed:
Refactor backDestination logic

Update `getPostCheckoutDestination` to return both destination and
backDestination. That way we're sharing the same logic for determing
whether this is the big sky flow or not.

p-jackson reviewed (approved):
I've rebased this onto `trunk` and refactored the logic into `getPostCheckoutDestination`. That way we're reusing the same logic for the checkout success and bail URLs.

TIL that you could simply go back to the plans step and it doesn't create a second site!
